ee/clickhouse/queries/funnels/funnel_unordered.py

```diff
@@ -109,13 +124,44 @@ def _get_step_times(self, max_steps: int):
 
     def get_sorting_condition(self, max_steps: int):
 
+        conditions = []
+
+        event_times_elements = []
+        for i in range(max_steps):
+            event_times_elements.append(f"latest_{i}")
+
+        conditions.append(f"arraySort([{','.join(event_times_elements)}]) as event_times")
+        # replacement of latest_i for whatever query part requires it, just like conversion_times
+
         basic_conditions: List[str] = []
         for i in range(1, max_steps):
             basic_conditions.append(
                 f"if(latest_0 < latest_{i} AND latest_{i} <= latest_0 + INTERVAL {self._filter.funnel_window_days} DAY, 1, 0)"
             )
 
+        conditions.append(f"arraySum([{','.join(basic_conditions)}, 1])")
+
         if basic_conditions:
-            return f"arraySum([{','.join(basic_conditions)}, 1])"
+            return ",".join(conditions)
         else:
             return "1"
+
+    def _get_exclusion_condition(self):
+        if not self._filter.exclusions:
+            return ""
+
+        conditions = []
+        for exclusion_id, exclusion in enumerate(self._filter.exclusions):
+            from_time = f"latest_{exclusion.funnel_from_step}"
+            to_time = f"event_times[{exclusion.funnel_to_step + 1}]"
+            exclusion_time = f"exclusion_{exclusion_id}_latest_{exclusion.funnel_from_step}"
+            condition = (
+                f"if( {exclusion_time} > {from_time} AND {exclusion_time} < "
```

Comment:
Is the "<" supposed to be trailing?

Reply:
multiple strings! These get concatenated automatically when enclosed in a paranthesis.

Felt clearer to me, about what's happening in the "<" (the next line)